File: go/space-age/space_age.go
package space

type Planet string

const EarthYearInSeconds float64 = 31557600

func Age(seconds float64, planet Planet) float64 {
	earthYears := seconds / EarthYearInSeconds

	switch planet {
	case "Mercury":
		return earthYears / 0.2408467
	case "Venus":
		return earthYears / 0.61519726
	case "Earth":
		return earthYears
	case "Mars":
		return earthYears / 1.8808158
	case "Jupiter":
		return earthYears / 11.862615
	case "Saturn":
		return earthYears / 29.447498
	case "Uranus":
		return earthYears / 84.016846
	case "Neptune":
		return earthYears / 164.79132
	default:
		return -1
	}
}

File: c/space-age/space_age.c
#include "space_age.h"

static float planet_multipliers[] = {
		0.2408467,	// Mercury
		0.61519726, // Venus
		1.0,				// Earth
		1.8808158,	// Mars
		11.862615,	// Jupiter
		29.447498,	// Saturn
		84.016846,	// Uranus
		164.79132,	// Neptune
};

float age(planet_t planet, int64_t seconds)
{
	if (planet < MERCURY || planet > NEPTUNE)
		return -1;

	float earth_years = seconds / 31557600;

	return earth_years / planet_multipliers[planet];
}